### PR TITLE
QA-235: pipeline: Switch to static checks by golangci-lint

### PR DESCRIPTION
**Pull request by lluiscampos**
And fix the code accordingly.

Bonus:

    Update dependencies: go mod tidy + go mod vendor

### CODE REVIEW

**Comment by lluiscampos** [comment]
FYI @mzedel

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :running_woman:

File: app/state.go [hunk 1]
```diff
@@ -1985,7 +2039,7 @@ func (c *controlMapState) Handle(ctx *StateContext, controller Controller) (Stat
 
 type fetchControlMapState struct {
 	baseState
-	update       *datastore.UpdateInfo
+	// update       *datastore.UpdateInfo
```
Comment (oleorhagen) [inline app/state.go:2042]: Q:
Leftover?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -17,10 +19,8 @@ test:
   image: golang:1.17
   before_script:
     - apt-get update && apt-get install -yyq liblzma-dev libssl-dev libglib2.0-dev dbus clang-format-9
-    - make get-tools
   script:
     - git ls-tree -r --name-only HEAD | grep -v vendor/ | grep '\.[ch]$' | xargs clang-format-9 -i
-    - make extracheck
```
Comment (oleorhagen) [inline .gitlab-ci.yml:23]: Should we remove the target from the makefile also?

File: dbus/error.go [hunk 3]
```diff
@@ -44,11 +44,3 @@ func (v *Error) Message() string {
 func ErrorFromNative(err Handle) error {
 	return &Error{C.to_error(unsafe.Pointer(err))}
 }
-
-// errorToNative returns an Error object from a native error
-func errorToNative(err error) Handle {
```
Comment (oleorhagen) [inline dbus/error.go:49]: :tada:

File: dbus/test/error_util.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//go:build !nodbus && cgo
+// +build !nodbus,cgo
+
+package test
```
Comment (oleorhagen) [inline dbus/test/error_util.go:17]: :muscle:

File: deb-requirements.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+liblzma-dev libssl-dev libglib2.0-dev
```
Comment (oleorhagen) [inline deb-requirements.txt:1]: :+1:

File: utils/kvparser.go [hunk 6]
```diff
@@ -52,11 +52,7 @@ func (k *KeyValParser) Parse(raw io.Reader) error {
 			return errors.Errorf("incorrect line '%s'", line)
 		}
 
-		if _, ok := k.data[val[0]]; ok {
```
Comment (oleorhagen) [inline utils/kvparser.go:55]: hah, fix looks a fair bit better :woman_cartwheeling:

**Review by oleorhagen — APPROVED** [review]
I had my jaw drop multiple times going through this!

So much better looking. Bug fixes, clean-ups, all through our code-base.

I'm lost for words. I think this is a great boon to our codebase!

Thank you so much for this, only the client part is a fair few story points!

:medal_military:  (of honour)

File: .gitlab-ci.yml (hunk 2 above)
Comment (lluiscampos) [reply .gitlab-ci.yml:23]: We could... but it is harmless to leave and maybe some of us are using it. I can remove it if you insist.

File: cli/setup.go [hunk 7]
```diff
@@ -27,15 +27,18 @@ import (
...
-	"github.com/mendersoftware/mender/client"
-	"github.com/mendersoftware/mender/conf"
-	"github.com/mendersoftware/mender/device"
-	"github.com/mendersoftware/mender/system"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/pkg/errors"
 	"github.com/urfave/cli/v2"
+
+	//nolint:staticcheck
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply cli/setup.go:35]: `golang.org/x/crypto/ssh/terminal` is deprecated in favor of `golang.org/x/term`. I have been doing these in two separate commits to have a clean changelog (see example [here](https://github.com/mendersoftware/tenantadm/pull/335/commits)) but on `mender` repo I just forgot to make the second commit :laughing: 

I'll fix now. See also my note to self above.

**Comment by lluiscampos** [comment]
> I had my jaw drop multiple times going through this!
> 
> So much better looking. Bug fixes, clean-ups, all through our code-base.
> 
> I'm lost for words. I think this is a great boon to our codebase!
> 
> Thank you so much for this, only the client part is a fair few story points!
> 
> medal_military (of honour)

Thank you very much for your words, highly appreciated :) TBH after doing every other repo, this one was a breeze :ninja:

**Comment by lluiscampos** [comment]
> @mender-test-bot start pipeline running_woman

I fixed the instrument patch, please try again :pager:

File: .gitlab-ci.yml (hunk 2 above)
Comment (oleorhagen) [reply .gitlab-ci.yml:23]: No insistence :)

**Comment by lluiscampos** [comment]
The [last force-push](https://github.com/mendersoftware/mender/compare/33b045826042fda5cd84898debe0a86f69d53bd8..86489f44a681026389a5f608b89191f5d9960a7e) only fixed a dependency for unit tests, the [previous integration pipeline](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/419582625) was green. Merging :tada:

**Comment by oleorhagen** [comment]
:tada: :tada:

**Pull request merged**